botnim/sync.py

```diff
@@ -51,116 +52,87 @@ def openapi_to_tools(openapi_spec):
-                          f'remaining {len(file_streams)}')
-                    file_streams = file_streams[32:]
-                vector_store_id = vector_store.id
-            tool_resources = dict(
-                file_search=dict(
-                    vector_store_ids=[vector_store_id],
-                ),
-            )
-        tools = [dict(
-            type='file_search',
-            file_search=dict(
-                max_num_results=context_.get('max_num_results', 20),
-            ),
-        )]
-
-    # List all the assistants in the organization:
-    assistants = client.beta.assistants.list()
-    assistant_id = None
-    assistant_name = config['name']
-    if not production:
-        assistant_name += ' - פיתוח'
-    for assistant in assistants:
-        if assistant.name == assistant_name:
-            assistant_id = assistant.id
-            break
-    print(f'Assistant ID: {assistant_id}')
-    asst_params = dict(
-        name=assistant_name,
-        description=config['description'],
-        model='gpt-4o',
-        instructions=config['instructions'],
-        temperature=0.00001,
-    )
+async def update_assistant(config, config_dir, production, replace_context=False, debug=False):
+    """Update or create an assistant with the given configuration
+    
+    Args:
+        config: Bot configuration
+        config_dir: Directory containing bot files
+        production: Whether this is a production environment
+        replace_context: If True, update both assistant and context. If False, only update assistant.
+        debug: If True, save downloaded files locally for debugging
+    """
+    # Initialize vector store backend and context manager
+    vs_backend = OpenAIVectorStore(production)
+    context_manager = ContextManager(config_dir, vs_backend)
+    
+    # Get the environment-specific assistant name first
+    assistant_name = context_manager._add_environment_suffix(config['name'])
+    
+    # Create new assistant parameters
+    assistant_params = {
+        'name': assistant_name,
+        'description': config['description'],
+        'model': 'gpt-4o',
+        'instructions': config['instructions'],
+        'temperature': 0.00001,
+    }
+    
+    # Handle tools configuration
+    tools = []
     if config.get('tools'):
-        tools = tools or []
         for tool in config['tools']:
             if tool == 'code-interpreter':
-                tools.append(dict(type='code_interpreter'))
+                tools.append({"type": "code_interpreter"})
             else:
                 openapi_spec = (SPECS / 'openapi' / tool).with_suffix('.yaml').open()
                 openapi_spec = yaml.safe_load(openapi_spec)
                 openapi_tools = openapi_to_tools(openapi_spec)
-                # print(f'OpenAPI Tool: {tool}')
                 tools.extend(openapi_tools)
+    
+    # Set up context and get tools configuration
+    tools_config = None
+    if config.get('context') and replace_context:
```

Comment:
If `replace_context` is not provided then the new/updated assistant will not have any vector stores attached.